IndicoToolkit/Association/Positioning.cs

```diff
@@ -0,0 +1,321 @@
+            }
+            else 
+            {
+                return 0f;
+            }
+        }
+
+        /// <summary>
+        /// Get the amount of vertical overlap between two bounding boxes.
+        /// </summary>
+        /// <param name="pos1">First position</param>
+        /// <param name="pos2">Second position</param>
+        public float getVerticalOverlap(Position pos1, Position pos2)
+        {
+            int pageDifference = Math.Abs(pos1.pageNum - pos2.pageNum);
+            if (pageDifference > 0)
+            {
+                throw new ToolkitInputException(
+                    "Predictions are not on the same page!"
+                );
+            }
+            if (yAxisOverlap(pos1, pos2))
+            {
+                float verticalOverlapDistance = Math.Abs(
+                    Math.Max(pos1.bbTop, pos2.bbTop) - Math.Min(pos1.bbBot, pos2.bbBot)
+                );
+                float positionHeight = Math.Abs(pos2.bbTop - pos2.bbBot);
+                return verticalOverlapDistance / positionHeight;
+            }
+            else 
+            {
+                return 0f;
+            }
+        }
+
+        /// <summary>
+        /// Get the vertical minimum distance between two bounding boxes.
+        /// </summary>
+        /// <param name="abovePos">The position expected to be above</param>
+        /// <param name="belowPos">The position expected to be below</param>
+        /// <param name="pageHeight">If you want to measure distances across pages, set the OCR page height otherwise locations on separate pages will raise an exception. Defaults to null.</param>
+        public float getVerticalMinDistance(Position abovePos, Position belowPos, int? pageHeight = null)
+        {
+            bool addPageHeight = false;
+            int pageDifference = Math.Abs(abovePos.pageNum - belowPos.pageNum);
+            if (pageDifference > 0)
+            {
+                if (pageHeight == null)
+                {
+                    throw new ToolkitInputException(
+                        "Predictions are not on the same page! Must enter a page height." 
+                    );
+                } 
+                else 
+                {
+                    addPageHeight = true;
+                }
+            }
+            float minDistance = belowPos.bbTop - abovePos.bbBot;
+            if (minDistance < 0)
+            {
+                minDistance = 0;
+            }
+            if (addPageHeight)
+            {
+                minDistance += (int) pageHeight * pageDifference;
+            }
+            return minDistance;
+        }
+
+        /// <summary>
+        /// Get the horizontal minimum distance between two bounding boxes.
+        /// </summary>
+        /// <param name="pos1">First position</param>
+        /// <param name="pos2">Second position</param>
+        public float getHorizontalMinDistance(Position pos1, Position pos2)
```

Comment:
should we be checking if the overlap for these methods and then returning 0?

Reply:
Yeah that makes sense. Can update docstring is say something like "returns 0 if bounding boxes are adjacent or overlap"